### PR TITLE
plumb staking_account and voting_keypair from multinode-demo to Vote

### PR DESCRIPTION
**Pull request by rob-solana**
#### Problem
 the vote account rework and removal of votesigner has left validators without any stake in multinode-demo
 setup

 #### Summary of Changes
 * build Pubkey from String
 * add staking_account and voting_pubkey to all the vote APIs
 * use wallet to airdrop to nodes, then to staking accounts,
 * plumb staking and voting through fullnode, tvu, replay_stage

### CODE REVIEW

File: core/src/fullnode.rs [hunk 1]
```diff
@@ -371,11 +379,13 @@ mod tests {
             GenesisBlock::new_with_leader(10_000, leader_keypair.pubkey(), 1000);
         let (validator_ledger_path, _blockhash) = create_new_tmp_ledger!(&genesis_block);
 
+        let keypair = Keypair::new();
```
Comment (mvines) [inline core/src/fullnode.rs]: nit: call this `voting_keypair`

File: fullnode/src/main.rs [hunk 2]
```diff
@@ -32,11 +32,18 @@ fn main() {
                 .help("File containing an identity (keypair)"),
         )
         .arg(
-            Arg::with_name("staker_keypair")
-                .long("staker-keypair")
+            Arg::with_name("staking_account")
+                .long("staking-account")
+                .value_name("PUBKEY_B58_STR")
```
Comment (mvines) [inline fullnode/src/main.rs]: nit: `PUBKEY_BASE58_STRING` -- don't make me guess!

File: multinode-demo/common.sh [hunk 3]
```diff
@@ -110,6 +110,69 @@ tune_system() {
...
+
+  # as the staker, set the node as the delegate and the staker as
+  #  the vote-signer
+  $solana_wallet --keypair "$staker_id_path" \
+                 configure-staking-account \
+                 --delegate-account "$fullnode_id" \
+                 --authorize-voter "$staker_id"  || return $?
+
+  return 0
+
```
Comment (mvines) [inline multinode-demo/common.sh]: nit: UNNECESSARY WHITESPACE!!1!

File: sdk/src/pubkey.rs [hunk 4]
```diff
@@ -2,11 +2,34 @@ use bs58;
...
 #[derive(Serialize, Deserialize, Clone, Copy, Default, Eq, PartialEq, Ord, PartialOrd, Hash)]
 pub struct Pubkey(GenericArray<u8, U32>);
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum ParsePubkeyError {
+    TooShort,
+    Invalid,
+}
+
+impl FromStr for Pubkey {
```
Comment (mvines) [inline sdk/src/pubkey.rs]: yno tests for this new sdk/ feature?  😿

File: fullnode/src/main.rs (hunk 2 above)
Comment (CriesofCarrots) [reply fullnode/src/main.rs]: We just call this plain old "PUBKEY" everywhere else

File: sdk/src/pubkey.rs [hunk 5]
```diff
@@ -2,11 +2,34 @@ use bs58;
...
+        let pubkey_vec = bs58::decode(s)
+            .into_vec()
+            .map_err(|_| ParsePubkeyError::Invalid)?;
+        if pubkey_vec.len() != mem::size_of::<Pubkey>() {
+            Err(ParsePubkeyError::TooShort)
+        } else {
+            Ok(Pubkey::new(&pubkey_vec))
+        }
+    }
+}
```
Comment (CriesofCarrots) [inline sdk/src/pubkey.rs]: ❤️ Thank you for this!

**Comment by rob-solana** [comment]
thanks!  missed it by 5 min.
